src/lib.rs

```diff
@@ -141,6 +144,28 @@ impl FixedBitSet
         }
     }
 
+    pub fn count_ones<T: IndexRange>(&self, range: T) -> usize
+    {
+        let start = range.start().unwrap_or(0);
+        let end = range.end().unwrap_or(self.length - 1);
+        // range makes sure that range.start <= range.end
+        assert!(end < self.length);
+        let (first_block, first_rem) = div_rem(start, BITS);
+        let (last_block, last_rem) = div_rem(end, BITS);
+        let mut sum = 0usize;
+        // we can't skip first_block in case first_block == last_block
+        for block in &self.data[first_block..last_block] {
+            sum += block.count_ones() as usize;
+        }
+        // calculate masks; deals with overflowing shr problem when x == 0
+        let mask = |x| if x != 0 { Block::max_value() >> (BITS - x) } else { 0 };
+        let mask_first_block: Block = mask(first_rem);
+        let mask_last_block: Block = mask(last_rem);
+        sum += (self.data[last_block] & mask_last_block).count_ones() as usize;
+        sum -= (self.data[first_block] & mask_first_block).count_ones() as usize;
```

Comment:
Can both of these masks be correct at the same time? One needs to include the start of a block, the other the end of it.